### ISSUE TITLE
No warning on unfulfilled `#[expect]`

### ISSUE THREAD

**Issue by lnicola** [rust-lang/rust-analyzer#18120]
```rust
#[expect(non_snake_case)]
fn main() {
    let _foo = 0;
}
```

The other half of #18056.

**Comment by ChayimFriedman2** [comment]
This will also be quite hard to achieve: if we have an `#[expect(...)] mod foo;` declaration in a file, we need to check all lints in `foo.rs` to determine if we should warn, which means checking `foo`'s children up to the entire crate. Since we don't want a diagnostics request to be that expensive, that means caching the diagnostics of a file, and the only way to cache between requests is in the DB. But if it's in the DB, it cannot be in crates/ide-diagnostics, and all handlers depend on IDE functionality...

**Comment by lnicola** [comment]
Yeah, it's fine to leave this for some later time which might never come. I filed it pre-emptively because I suspect people will complain anyway.

**Comment by Veykril** [comment]
Imo this is something we shouldn't even bother implementing. Any small divergence between r-a and rustc could make us emit a false diagnostic from this. "Having to" fallback to cargo for this to be checked seem absolutely fine in my eyes as this is purely a check that your `allow` is actually doing something.
